src/ServiceControl.Transports.ASBS/CheckDeadLetterQueue.cs

```diff
@@ -0,0 +1,46 @@
+﻿namespace ServiceControl.Transports.ASBS
+{
+    using System;
+    using System.Configuration;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.ServiceBus.Management;
+    using NServiceBus.CustomChecks;
+    using NServiceBus.Logging;
+
+    public class CheckDeadLetterQueue : CustomCheck
+    {
+        public CheckDeadLetterQueue(TransportSettings settings) : base(id: "Dead Letter Queue", category: "Transport", repeatAfter: TimeSpan.FromHours(1))
+        {
+            Logger.Debug("Azure Service Bus Dead Letter Queue custom check starting");
+
+            var connectionStringSettings = ConfigurationManager.ConnectionStrings["NServiceBus/Transport"];
+            var transportConnectionString = connectionStringSettings.ConnectionString;
+            managementClient = new ManagementClient(transportConnectionString);
+            stagingQueue = $"{settings.EndpointName}.staging";
+        }
+
+        public async override Task<CheckResult> PerformCheck()
+        {
+            Logger.Debug("Checking Dead Letter Queue length");
+
+            var queueRuntimeInfo = await managementClient.GetQueueRuntimeInfoAsync(stagingQueue).ConfigureAwait(false);
+            var deadLetterMessageCount = queueRuntimeInfo.MessageCountDetails.DeadLetterMessageCount;
```

Comment:
What happens when this fails?

Reply:
The custom check infra will try catch and mark it as a failure

https://github.com/Particular/NServiceBus.CustomChecks/blob/develop/src/NServiceBus.CustomChecks/TimerBasedPeriodicCheck.cs#L38